test/core/unit/engine_http.test.js

```diff
@@ -247,21 +435,152 @@ test('hooks - afterResponse', (t) => {
       t.fail();
     }
 
-    t.assert(finalContext.answer === answer);
+    t.ok(finalContext.answer === answer);
 
     t.end();
   });
 });
 
-test('Redirects', (t) => {
-  const target = nock('http://localhost:8888')
-    .get('/foo')
-    .reply(302, undefined, {
-      Location: '/bar'
-    })
-    .get('/bar')
-    .reply(200, { foo: 'bar' });
+test('hooks - beforeScenario', (t) => {
+  const endpoint = '/products';
+  const script = {
+    config: {
+      target: 'http://localhost:8888',
+      processor: {
+        setEndpoint: function (context, ee, next) {
+          context.vars.endpoint = endpoint;
+
+          t.ok(
```

Comment:
Is the goal here to check the signature of hook functions?

If so, should we have some assertions on other parameters too?